frontend/src/metabase/visualizations/visualizations/SmartScalar/compute.js

```diff
@@ -431,14 +439,14 @@ function getRowOfPeriodsAgo({
   // skip the latest element since that is our current value
   const searchIndexStart = nextValueRowIndex - 1;
   if (searchIndexStart < 0) {
-    return -1;
+    return undefined;
```

Comment:
This was inconsistent with the rest of the functions return types. It could be causing errors for some people who look at dates in the past that are close to the start of their dataset, since the consuming function thinks it will either have a value or be `undefined`. `-1` is neither of those.